src/components/form-component.js

```diff
@@ -329,6 +339,8 @@ function createFormClass(s = defaultStrategy) {
         s.actions.addIntent(model, { type: 'submit' }),
       ]));
 
+      if (onSubmitFailed) onSubmitFailed();
```

Comment:
Caveat: I'm not familiar with this code, so it's possible this line shouldn't be there.

The desired behavior is: whenever a submit occurs, either `onSubmit` or `onSubmitFailed` should be called.  This line seemed necessary to achieve that behavior.

`onSubmit` gets called here: https://github.com/dave-clover/react-redux-form/blob/d6547e8bdff31419558b7199e373a1714d4dc1a2/src/components/form-component.js#L288 , so it seemed like `onSubmitFailed` also needed to be in this meeting.

I'll take another look and write some tests around it to ensure it's doing what it should be doing.